src/v/redpanda/admin_server.h

```diff
@@ -96,6 +96,29 @@ class admin_server {
         return auth_state;
     }
 
+    void log_exception(
+      const ss::sstring& url,
+      const request_auth_result& auth_state,
+      std::exception_ptr eptr) const;
+
+    /**
+     * Helper for intercepting all exceptions thrown by all request handlers for
+     * the intention of logging them. If intercepted and logged the exception
+     * will be rethrown and handled by the http server.
+     * */
+    template<typename F>
+    auto intercept_exceptions(
+      const ss::sstring& url,
+      const request_auth_result& auth_state,
+      F handler) {
+        try {
+            return handler();
+        } catch (...) {
```

Comment:
handler returns a future, so I think you'd need to co_await it here and then co_return the return type, in order to actually get exceptions thrown from any part of the handler past the first co_await.

Reply:
Ah nice catch, I actually introduced this bug attempting a refactor